rental_manager.rb

```diff
@@ -0,0 +1,55 @@
+require_relative 'rental'
+
+class RentalManager
+  def initialize
+    @rentals = []
+  end
+
+  # Create a rental
+  def create_rental(people_manager, book_manager)
+    date, person_index, book_index = rental_data(people_manager, book_manager)
+
+    @rentals << Rental.new(date, people_manager.people[person_index], book_manager.books[book_index])
+    puts 'Rental created successfully'
+  end
+
+  def rental_data(people_manager, book_manager)
+    puts 'Select a book from the following list by number: '
+    book_manager.list_all_books
+    book_index = gets.chomp.to_i
+
+    puts 'Select a person from the following list by number (Not ID): '
+    people_manager.list_all_people
+    person_index = gets.chomp.to_i
+
+    print 'Date: '
+    date = gets.chomp
+
+    [date, person_index, book_index]
+  end
+
+  # List all rentals for a given person id
+  def list_rentals_by_person_id
+    id, rentals = rentals_by_id
+
+    if rentals.empty?
+      puts 'This person has no rentals'
+    else
+      puts "Rentals for person with ID #{id}:"
+      rentals.each do |rental|
+        puts "Date: #{rental.date}, Book: \"#{rental.book.title}\" by #{rental.book.author}"
+      end
+    end
+  end
```

Comment:
- [ ] @NicolasAndrehh and @simplegoose great work in refactoring your code, however, I would suggest you move this function to another file, according to the `Single-responsibility Principle` each file should have a single responsibility this will help minimize errors debug quickly and also scale up the app easier in future, in our case all other methods handle inputs but this method does not.

Reply:
That method handles the id input using the rentals_by_id method, or do you mean something else?